test/gtest/conv_embed_db.cpp

```diff
@@ -0,0 +1,215 @@
+#include <tuple>
+
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/env.hpp>
+#include "conv_2d.hpp"
+#include "get_handle.hpp"
+
+std::string GetFloatArg()
+{
+    static const auto tmp = miopen::GetEnv("MIOPEN_TEST_FLOAT_ARG");
+    if(tmp.empty())
+    {
+        return "";
+    }
+    return tmp.front();
+};
+
+void GetArgs(const std::string& param, std::vector<std::string>& tokens)
+{
+    std::stringstream ss(param);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+class ConfigWithHalf : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithInt8 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithBFloat16 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithFloat : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+
+    std::vector<std::string> params;
+    switch(prec)
+    {
+    case miopenFloat: params = ConfigWithFloat::GetParam(); break;
+    case miopenHalf: params = ConfigWithHalf::GetParam(); break;
+    case miopenInt8: params = ConfigWithInt8::GetParam(); break;
+    case miopenBFloat16: params = ConfigWithBFloat16::GetParam(); break;
+    case miopenInt8x4:
+    case miopenInt32:
+    case miopenDouble:
+        MIOPEN_THROW(miopenStatusBadParm,
+                     "miopenInt8x4, miopenInt32, miopenDouble data type not supported by "
+                     "conv_embed_db test");
+
+    default: params = ConfigWithFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(), tokens.end(), std::back_inserter(ptrs), [](const auto& str) {
+            return str.data();
+        });
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        EXPECT_FALSE(capture.find("Perf Db: record not found") != std::string::npos);
+    }
+};
+
+TEST_P(ConfigWithFloat, FloatTest)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if(miopen::StartsWith(handle.GetDeviceName(), "gfx908") || // Explicitly disabled
+       miopen::StartsWith(handle.GetDeviceName(), "gfx90a") || // Explicitly disabled
+       miopen::StartsWith(handle.GetDeviceName(), "gfx94") ||  // Implicitly disabled by default
+       miopen::StartsWith(handle.GetDeviceName(), "gfx103") || // Implicitly disabled by default
+       miopen::StartsWith(handle.GetDeviceName(), "gfx110") || // Implicitly disabled by default
+       GetFloatArg() != "--float")
+    {
+        GTEST_SKIP();
+    }
+    else
+    {
+        Run2dDriver(miopenFloat);
+    }
```

Comment:
[R] This code reproduces the details of implementation of test/CMakeLists.txt.

AFAIU this better reflects what the developer of the test wants (he/she wants to test on gfx906/900 only):
```suggestion
    if((handle.GetDeviceName() == "gfx906" || handle.GetDeviceName() == "gfx900") //
       && GetFloatArg() == "--float")
    {
        Run2dDriver(miopenFloat);
    }
    else
    {
        GTEST_SKIP();
    }
```

Reply:
Here's the original test, GFX908 and GFX90A are disabled explicitly. 
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/2497e64ce5fa0ee562a9fdadc19273de86f6d618/test/CMakeLists.txt#L773

Yet, this list is not complete because of these default setting, GFX94X_DISABLED, GFX103X_DISABLED, GFX110X_DISABLED. Thus, the comments "Implicitly disabled by default".

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/2b656e53fd217526dab15e307f9e99c8dc045adb/test/CMakeLists.txt#L518

Reply:
The code should reflect the intent of the developer wherever possible.

Reply:
> The code should reflect the intent of the developer wherever possible.

I was too rush to make that comment. Your proposed change is indeed cleaner.

Reply:
[Resolved]